test/requests/formatting_test.rb

```diff
@@ -0,0 +1,54 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test_helper"
+
+class FormattingTest < Minitest::Test
+  def setup
+    @document = RubyLsp::Document.new(+<<~RUBY)
+      class Foo
+      def foo
+      end
+      end
+    RUBY
+  end
+
+  def test_formats_with_rubocop_when_present
+    assert_equal(<<~RUBY, formatted_document)
+      # typed: true
+      # frozen_string_literal: true
+
+      class Foo
+        def foo
+        end
+      end
+    RUBY
+  end
+
+  def test_formats_with_syntax_tree_when_rubocop_is_not_present
+    rubocop_paths = $LOAD_PATH.select { |path| path.include?("gems/rubocop") }
```

Comment:
Nice test, but could we encapsulate the bits around the actual test into something like:
```ruby
def with_disabled_rubocop(&blk)
  # Do the LOAD_PATH magic
  blk.call
ensure
  # Do cleanup
end
```
so that the test itself reads more meaningfully?